### DISCUSSION TITLE
[Question][Hw2] Variable shadowing in program equivalence checker

### DISCUSSION THREAD

**Discussion by chlee973** [prosyslab-classroom/cs424-program-reasoning#424] [Q&A] [answered]
### Your Name

이창해

### Question

Can I assume that programs using variable shadowing will not be given as input when implementing a program equivalence checker? I’m concerned that issues might arise with duplicate variable names while implementing the logic related to them.
e.g.
(let y (+ x 2) (let y (+ x 1) y))

**Comment by spearo2** [comment, marked as answer]
Thank you for your question.
The homework instruction actually has the answer for your question.

```
Assume that functions only have one parameter named x and each variable has a unique name.
```

**Reply by spearo2** [reply]
closing it as it was marked answered

**Comment by KihongHeo** [comment]
한국어로 질문하세요. 전문가들은 올바른 국어로 공부하는 연습도 필요합니더. 그리고 chatgpt로 번역해서 아래에 붙여주세요.

Github 심지어 자동으로 번역도 된답니다.

**Reply by chlee973** [reply]
감사합니다. 다음부터는 한국어 문장과 영어 문장을 함께 적어 올리도록 하겠습니다.
